src/components/tasks/WithDevPassword.js

```diff
@@ -3,17 +3,19 @@ import { useDevPassword } from '../../utils/hooks/useDevPassword';
 import { useParams } from 'react-router-dom'
 import { TasksPassword } from './TasksPassword';
 
-export const WithDevPassword = ({ WrappedComponent, wrappedProps, createPassword }) => {
-    const {id} = useParams()
+export const WithDevPassword = ({ WrappedComponent, wrappedProps, createPassword, text }) => {
 
+    const params = useParams()
+    
     const [password, setPassword] = useState("")
-    const { set } = useDevPassword()
+    console.log(password)
+    const { set, get } = useDevPassword()
 
     const handleSubmit = (password) => {
-        set(createPassword(id, password))
-        console.log(createPassword(id, password))
+        set(createPassword(params.id, password))
         setPassword(password)
     }
 
-    return !!password ? <WrappedComponent {...wrappedProps} id={id} /> : <TasksPassword handleSubmit={handleSubmit}/>
-}
\ No newline at end of file
+    return !!password || get() ? <WrappedComponent {...wrappedProps} id={params.id} {...params} /> : <TasksPassword handleSubmit={handleSubmit} text={text}/>
```

Comment:
Dlaczego tu robimy podwójną negację XD A jeśli miałaby być pojedyncza to nie wiem czy nie obróciłbym warunków ternariuszu. Żeby nie musieć obracać logicznie warunków.

Reply:
yyyy nie jest to podwójna negacja XDDD Ale to jest js - pojedyncze ! castuje do przeciwnego booleana, a !! kastuje do booleana. Tak samo, jak możesz zrobić chyba `+'3'` i to jest wtedy liczba 3.

Reply:
Znaczy to dosyć mocno wydaje mi się że jest to podwójna negacja XDD. Tylko w jsie pewnie służy do tego żeby zamienić wartość do bool-a, bo mają kiepski duck-typing. Ale to jest dziwne że żeby zrobić ewaluacje do boola. To najpierw robisz negacje i kastuje ci się to do boola, a potem wracasz do poprawnej wartości przez kolejną negację XDDD

Reply:
Ogólnie da się z tego co wiem castować inaczej, ale po co jak możesz napisać dwa wykrzykniki xd